torch/lib/c10d/NCCLUtils.cpp

```diff
@@ -10,14 +10,17 @@ std::string getNcclVersion() {
   std::call_once(ncclGetVersionFlag, []() {
     int version;
     ncclResult_t status = ncclGetVersion(&version);
-    if (status != ncclSuccess) {
+    // can't compute the version if call did not return successfully or version
+    // code < 100 (corresponding to 0.1.0)
```

Comment:
code -> version?

Any reason for this? Does this mean if `version == 90`, `ncclGetVersion` would return success but it actually failed? BTW, the earliest version I can find on github is 1.2.1, when would we see code < 100?

Reply:
@mrshenli Yes that's right, I don't think we'll ever see code < 100 (1.2.1 corresponds to 1201 for example). Right now in the implementation of this function, (https://github.com/NVIDIA/nccl/blob/ccb1298148327bacb9b83452ed6ae0b29417e7e2/src/init.cc#L173) a non-success code is only returned if we pass in an invalid argument. I just want to be extra-careful though and guard against all possible errors here, as if we don't do this then we could have divide by 0 issues in the code below.